src/bin/lttng-relayd/cmd-2-11.c

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (C) 2013 - Jonathan Rajotte <jonathan.rajotte-julien@efficios.com>
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License, version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#define _LGPL_SOURCE
+#include <assert.h>
+
+#include <common/common.h>
+#include <common/sessiond-comm/relayd.h>
+
+#include <common/compat/endian.h>
+#include <common/compat/string.h>
+#include <lttng/constant.h>
+
+#include "cmd-2-11.h"
+#include "lttng-relayd.h"
+
+int cmd_create_session_2_11(const struct lttng_buffer_view *payload,
+		char *session_name, char *hostname,
+		uint32_t *live_timer, bool *snapshot)
+{
+	int ret;
+	struct lttcomm_relayd_create_session_2_11 header;
+	size_t header_len, received_names_size;
+	struct lttng_buffer_view session_name_view;
+	struct lttng_buffer_view hostname_view;
+
+	header_len = sizeof(header);
+
+	if (payload->size < header_len) {
+		ERR("Unexpected payload size in \"cmd_create_session_2_11\": expected >= %zu bytes, got %zu bytes",
+				header_len, payload->size);
+		ret = -1;
+		goto error;
+	}
+	memcpy(&header, payload->data, header_len);
+
+	header.session_name_len = be32toh(header.session_name_len);
+	header.hostname_len = be32toh(header.hostname_len);
+	header.live_timer = be32toh(header.live_timer);
+	header.snapshot = be32toh(header.snapshot);
+
+	received_names_size = header.session_name_len + header.hostname_len;
+	if (payload->size < header_len + received_names_size) {
+		ERR("Unexpected payload size in \"cmd_create_session_2_11\": expected >= %zu bytes, got %zu bytes",
+				header_len + received_names_size, payload->size);
+		ret = -1;
+		goto error;
+	}
+
+	/* Validate length against defined constant. */
+	if (header.session_name_len > LTTNG_NAME_MAX) {
+		ret = -ENAMETOOLONG;
+		ERR("Session name too long");
+		goto error;
+	}
+	if (header.hostname_len > LTTNG_HOST_NAME_MAX) {
+		ret = -ENAMETOOLONG;
+		ERR("Session name too long");
+		goto error;
+	}
+
+	/* Validate that names are (NULL terminated. */
+	session_name_view = lttng_buffer_view_from_view(payload, header_len,
+			    header.session_name_len);
```

Comment:
There are stray spaces here (after the 3 tabs).

Reply:
done